pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -1440,6 +1440,22 @@ type GatewaySpec struct {
 	HostNetwork *bool `json:"hostNetwork,omitempty"`
 }
 
+// EndpointAddress is a tuple that describes a single IP address or host name. This is a subset of
+// Kubernetes's v1.EndpointAddress.
+// +structType=atomic
+type EndpointAddress struct {
+	// The IP of this endpoint.
+	// May not be loopback (127.0.0.0/8), link-local (169.254.0.0/16),
+	// or link-local multicast ((224.0.0.0/24).
+	// IPv6 is also accepted but not fully supported on all platforms. Also, certain
+	// kubernetes components, like kube-proxy, are not IPv6 ready.
+	// TODO: This should allow hostname or IP, See #4447.
```

Comment:
Is this TODO still needed? There is a separate IP and hostname field already. #4447 doesn't look related either.